Remotion/Core/Extensions.UnitTests/Logging/Log4NetTests/Log4NetLoggerTest.cs

```diff
@@ -0,0 +1,154 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using log4net.Appender;
+using log4net.Core;
+using log4net.Layout;
+using log4net.Repository.Hierarchy;
+using NUnit.Framework;
+using Remotion.Development.NUnit.UnitTesting;
+using Remotion.Logging.Log4Net;
+using Remotion.Reflection;
+using MicrosoftLogging = Microsoft.Extensions.Logging;
+
+namespace Remotion.Extensions.UnitTests.Logging.Log4NetTests;
+
+public class Log4NetLoggerTest
+{
+  private TextWriter _originalConsoleOut;
+  private StringWriter _fakeConsoleOut;
+
+  [SetUp]
+  public void SetUp ()
+  {
+    _originalConsoleOut = Console.Out;
+    _fakeConsoleOut = new StringWriter();
+    Console.SetOut(_fakeConsoleOut);
+
+  }
+
+  [TearDown]
+  public void TearDown ()
+  {
+    log4net.LogManager.ResetConfiguration();
+    Console.SetOut(_originalConsoleOut);
+  }
+
+  [Test]
+  public void Convert_WithValidLogLevel_ReturnsConvertedLogLevel ()
```

Comment:
I think this could also be done using test cases, which would probably lead to a more readable test.

We should also split the act and assert parts of this test.